tests-integration/tests/cosmwasm/fixture.rs

```diff
@@ -138,56 +138,44 @@ impl Fixture {
     }
 
     pub fn verify_client_message(&self, deps: Deps<'_>, client_message: Vec<u8>) {
-        let resp = self
+        let resp: VerifyClientMessageResponse = self
             .query(
                 deps,
                 VerifyClientMessageRaw {
                     client_message: client_message.into(),
-                }
-                .into(),
+                },
             )
+            .and_then(from_json)
             .unwrap();
 
         assert!(resp.is_valid);
-        assert!(resp.status.is_none());
-        assert!(resp.found_misbehaviour.is_none());
     }
 
     pub fn check_for_misbehaviour(&self, deps: Deps<'_>, client_message: Vec<u8>) {
-        let resp = self
+        let resp: CheckForMisbehaviourResponse = self
             .query(
                 deps,
                 CheckForMisbehaviourMsgRaw {
                     client_message: client_message.into(),
-                }
-                .into(),
+                },
             )
+            .and_then(from_json)
             .unwrap();
 
-        assert!(resp.is_valid);
-        assert_eq!(resp.found_misbehaviour, Some(true));
+        assert!(resp.found_misbehaviour);
     }
 
     pub fn check_client_status(&self, deps: Deps<'_>, expected: Status) {
-        let resp = self.query(deps, StatusMsg {}.into()).unwrap();
-
-        assert_eq!(resp.status, Some(expected.to_string()));
-    }
+        let resp: StatusResponse = self.query(deps, StatusMsg {}).and_then(from_json).unwrap();
 
-    pub fn get_metadata(&self, deps: Deps<'_>) -> Option<Vec<GenesisMetadata>> {
-        self.query(deps, ExportMetadataMsg {}.into())
-            .map(|resp| resp.genesis_metadata)
-            .unwrap()
+        assert_eq!(resp.status, expected.to_string());
     }
 
-    pub fn query(&self, deps: Deps<'_>, msg: QueryMsg) -> StdResult<QueryResponse> {
+    pub fn query(&self, deps: Deps<'_>, msg: impl Into<QueryMsg>) -> StdResult<Binary> {
```

Comment:
Can we use a generic type as follow, and remove the `.and_then(from_json)` call on the query result?

```rust
    pub fn query<T: DeserializeOwned>(
        &self,
        deps: Deps<'_>,
        msg: impl Into<QueryMsg>,
    ) -> StdResult<T> {...}
```

Reply:
I did this initially, however, the tests didn't pass with no default features because `serde` is an optional dependency (so `serde::DeserializeOwned` can't be imported) whereas `cosmwasm_std` is not.

Reply:
Hmm. In fact, the entire cosmwasm tests should be put under the `std` feature since they only work with `std`. Let's leave it as is for now, nothing to worry about. Thanks.